commit 97aa593a8387586095b7eac12974ba2fdd08f4c3
Author: Johannes Doerfert <johannes@jdoerfert.de>
Date:   Mon Apr 6 12:39:40 2020 -0500

    [OpenMP] Fix layering problem with FrontendOpenMP
    
    Summary:
    ASTMatchers is used in various places and it now exposes the
    LLVMFrontendOpenMP library to its users without them needing to depend
    on it explicitly.
    
    Reviewers: lebedev.ri
    
    Subscribers: mgorny, yaxunl, bollu, guansong, martong, cfe-commits
    
    Tags: #clang
    
    Differential Revision: https://reviews.llvm.org/D77574

Index: llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang-tools-extra/clang-reorder-fields/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot-11~++20200407100707+717a4608676.orig/clang-tools-extra/clang-reorder-fields/CMakeLists.txt
+++ llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang-tools-extra/clang-reorder-fields/CMakeLists.txt
@@ -1,4 +1,5 @@
 set(LLVM_LINK_COMPONENTS
+  FrontendOpenMP
   support
 )
 
Index: llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang-tools-extra/clang-tidy/openmp/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot-11~++20200407100707+717a4608676.orig/clang-tools-extra/clang-tidy/openmp/CMakeLists.txt
+++ llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang-tools-extra/clang-tidy/openmp/CMakeLists.txt
@@ -1,4 +1,5 @@
 set(LLVM_LINK_COMPONENTS
+  FrontendOpenMP
   Support)
 
 add_clang_library(clangTidyOpenMPModule
Index: llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang/lib/ASTMatchers/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot-11~++20200407100707+717a4608676.orig/clang/lib/ASTMatchers/CMakeLists.txt
+++ llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang/lib/ASTMatchers/CMakeLists.txt
@@ -1,6 +1,7 @@
 add_subdirectory(Dynamic)
 
 set(LLVM_LINK_COMPONENTS
+  FrontendOpenMP
   Support
 )
 
@@ -14,5 +15,3 @@ add_clang_library(clangASTMatchers
   clangBasic
   clangLex
   )
-
-target_link_libraries(clangASTMatchers PUBLIC LLVMFrontendOpenMP)
Index: llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang/lib/StaticAnalyzer/Checkers/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot-11~++20200407100707+717a4608676.orig/clang/lib/StaticAnalyzer/Checkers/CMakeLists.txt
+++ llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang/lib/StaticAnalyzer/Checkers/CMakeLists.txt
@@ -1,4 +1,5 @@
 set(LLVM_LINK_COMPONENTS
+  FrontendOpenMP
   Support
   )
 
Index: llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang/lib/StaticAnalyzer/Core/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot-11~++20200407100707+717a4608676.orig/clang/lib/StaticAnalyzer/Core/CMakeLists.txt
+++ llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang/lib/StaticAnalyzer/Core/CMakeLists.txt
@@ -1,4 +1,5 @@
 set(LLVM_LINK_COMPONENTS
+  FrontendOpenMP
   Support
 )
 
Index: llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang/lib/Tooling/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot-11~++20200407100707+717a4608676.orig/clang/lib/Tooling/CMakeLists.txt
+++ llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang/lib/Tooling/CMakeLists.txt
@@ -1,5 +1,6 @@
 set(LLVM_LINK_COMPONENTS
   Option
+  FrontendOpenMP
   Support
   )
 
Index: llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang/lib/Tooling/Transformer/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot-11~++20200407100707+717a4608676.orig/clang/lib/Tooling/Transformer/CMakeLists.txt
+++ llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang/lib/Tooling/Transformer/CMakeLists.txt
@@ -1,4 +1,5 @@
 set(LLVM_LINK_COMPONENTS
+  FrontendOpenMP
   Support
 )
 
Index: llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang/unittests/AST/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot-11~++20200407100707+717a4608676.orig/clang/unittests/AST/CMakeLists.txt
+++ llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang/unittests/AST/CMakeLists.txt
@@ -1,4 +1,5 @@
 set(LLVM_LINK_COMPONENTS
+  FrontendOpenMP
   Support
   )
 
Index: llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang/unittests/ASTMatchers/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot-11~++20200407100707+717a4608676.orig/clang/unittests/ASTMatchers/CMakeLists.txt
+++ llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang/unittests/ASTMatchers/CMakeLists.txt
@@ -1,4 +1,5 @@
 set(LLVM_LINK_COMPONENTS
+  FrontendOpenMP
   Support
   )
 
Index: llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang/unittests/ASTMatchers/Dynamic/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot-11~++20200407100707+717a4608676.orig/clang/unittests/ASTMatchers/Dynamic/CMakeLists.txt
+++ llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang/unittests/ASTMatchers/Dynamic/CMakeLists.txt
@@ -1,4 +1,5 @@
 set(LLVM_LINK_COMPONENTS
+  FrontendOpenMP
   Support
   )
 
Index: llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang/unittests/Analysis/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot-11~++20200407100707+717a4608676.orig/clang/unittests/Analysis/CMakeLists.txt
+++ llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang/unittests/Analysis/CMakeLists.txt
@@ -1,4 +1,5 @@
 set(LLVM_LINK_COMPONENTS
+  FrontendOpenMP
   Support
   )
 
Index: llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang/unittests/Rename/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot-11~++20200407100707+717a4608676.orig/clang/unittests/Rename/CMakeLists.txt
+++ llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang/unittests/Rename/CMakeLists.txt
@@ -1,4 +1,5 @@
 set(LLVM_LINK_COMPONENTS
+  FrontendOpenMP
   support
   )
 
Index: llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang/unittests/Sema/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot-11~++20200407100707+717a4608676.orig/clang/unittests/Sema/CMakeLists.txt
+++ llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang/unittests/Sema/CMakeLists.txt
@@ -1,4 +1,5 @@
 set(LLVM_LINK_COMPONENTS
+  FrontendOpenMP
   Support
   )
 
Index: llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang/unittests/StaticAnalyzer/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot-11~++20200407100707+717a4608676.orig/clang/unittests/StaticAnalyzer/CMakeLists.txt
+++ llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang/unittests/StaticAnalyzer/CMakeLists.txt
@@ -1,4 +1,5 @@
 set(LLVM_LINK_COMPONENTS
+  FrontendOpenMP
   Support
   )
 
Index: llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang/unittests/Tooling/CMakeLists.txt
===================================================================
--- llvm-toolchain-snapshot-11~++20200407100707+717a4608676.orig/clang/unittests/Tooling/CMakeLists.txt
+++ llvm-toolchain-snapshot-11~++20200407100707+717a4608676/clang/unittests/Tooling/CMakeLists.txt
@@ -1,5 +1,6 @@
 set(LLVM_LINK_COMPONENTS
   ${LLVM_TARGETS_TO_BUILD}
+  FrontendOpenMP
   Support
   )
 
